code/game/objects/items/devices/radio/radio.dm

```diff
@@ -164,15 +171,21 @@
 
 		return
 
-	else if (href_list["freq"])
-		var/new_frequency = (frequency + text2num(href_list["freq"]))
-		if (!freerange || (frequency < 1200 || frequency > 1600))
+	else if("set_freq" in href_list)
+		new_frequency=frequency
+		if(href_list["set_freq"]!="-1")
+			new_frequency = text2num(href_list["set_freq"])
+		else
+			new_frequency = input(usr, "Set a new frequency (1200-1600 kHz).", src, frequency) as null|num
 			new_frequency = sanitize_frequency(new_frequency, maxf)
+			set_frequency(new_frequency)
+		check_traitor_uplink(frequency)
+
+	else if (href_list["freq"])
+		new_frequency = (frequency + text2num(href_list["freq"]))
+		new_frequency = sanitize_frequency(new_frequency, maxf)
 		set_frequency(new_frequency)
-		if(hidden_uplink)
-			if(hidden_uplink.check_trigger(usr, frequency, traitor_frequency))
-				usr << browse(null, "window=radio")
-				return
+		check_traitor_uplink(frequency)
```

Comment:
This is not equivalent to the old behavior, because now this proc doesn't `return` if `hidden_uplink` is true
I don't know if this matters
if you change this change the other call to `check_traitor_uplink` as well

Reply:
Would adding another `return` in the proc fix it? It seems to work as is.

Reply:
trailing returns do nothing
you should return a value like 1 and check for that value at the call site
if that returned value is 1 then you return from the calling proc

Reply:
See latest change, is this what you mean?